src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -116,6 +116,11 @@ internal static BoundStatement BindUsingStatementOrDeclarationFromParts(SyntaxNo
             {
                 expressionOpt = usingBinderOpt!.BindTargetExpression(diagnostics, originalBinder);
                 hasErrors |= !bindDisposable(fromExpression: true, out patternDisposeInfo, out awaitableTypeOpt);
+                Debug.Assert(expressionOpt is not null);
+                if (expressionOpt.Type is not null)
+                {
+                    CheckRestrictedTypeInAsyncMethod(originalBinder.ContainingMemberOrLambda, expressionOpt.Type, diagnostics, expressionOpt.Syntax);
```

Comment:
>CheckRestrictedTypeInAsyncMethod(originalBinder.ContainingMemberOrLambda, expressionOpt.Type, diagnostics, expressionOpt.Syntax); [](http://example.com/codeflow?start=20&length=129)

Is the wording of the error going to be confusing here?There is no user declared local here